php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/PHPStanReportParser.java

```diff
@@ -0,0 +1,204 @@
+        if (!readAllLines.isEmpty() && readAllLines.get(0).startsWith("<?xml")) { // NOI18N
+            return;
+        }
+        for (String line : readAllLines) {
+            if (!content) {
+                if (line.startsWith("<?xml")) { // NOI18N
+                    content = true;
+                }
+                continue;
+            }
+            if (content) {
+                newLines.add(line);
+                if (line.equals("</checkstyle>")) { // NOI18N
+                    break;
+                }
+            }
+        }
+        Files.write(Paths.get(fileName), newLines, StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+        if ("file".equals(qName)) { // NOI18N
+            processFileStart(attributes);
+        } else if ("error".equals(qName)) { // NOI18N
+            processResultStart(attributes);
+        }
+    }
+
+    @Override
+    public void endElement(String uri, String localName, String qName) throws SAXException {
+        if ("file".equals(qName)) { // NOI18N
+            processFileEnd();
+        } else if ("error".equals(qName)) { // NOI18N
+            processResultEnd();
+        }
+    }
+
+    private void processFileStart(Attributes attributes) {
+        assert currentResult == null : currentResult.getFilePath();
+        assert currentFile == null : currentFile;
+
+        currentFile = getCurrentFile(attributes.getValue("name")); // NOI18N
+    }
+
+    private void processFileEnd() {
+        assert currentFile != null;
+        currentFile = null;
+    }
+
+    private void processResultStart(Attributes attributes) {
+        assert currentFile != null;
+        assert currentResult == null : currentResult.getFilePath();
+
+        currentResult = new Result(currentFile);
+        int lineNumber = getInt(attributes, "line"); // NOI18N
+        // line number can be 0
+        // e.g. <error line="0" column="1" severity="error"
+        // message="Class PhpParser\Builder\ClassTest was not found while trying to analyse it - autoloading is probably not configured properly." />
+        if (lineNumber == 0) {
+            lineNumber = 1;
+        }
+        currentResult.setLine(lineNumber);
+        currentResult.setColumn(getInt(attributes, "column")); // NOI18N
+        String message = attributes.getValue("message"); // NOI18N
+        currentResult.setCategory(String.format("%s: %s", attributes.getValue("severity"), message)); // NOI18N
+        currentResult.setDescription(message);
+    }
+
+    private void processResultEnd() {
+        assert currentResult != null;
+        results.add(currentResult);
+        currentResult = null;
+    }
+
+    private int getInt(Attributes attributes, String name) {
+        int i = -1;
+        try {
+            i = Integer.parseInt(attributes.getValue(name));
```

Comment:
Nitpick: `Integer.valueOf()` is a bit better of possible internal cache usage.

Reply:
Will fix it.